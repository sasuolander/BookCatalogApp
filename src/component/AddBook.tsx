import React from 'react'

export const AddBook =({})=>{
    return(<div><p>test</p><form></form></div>)
}