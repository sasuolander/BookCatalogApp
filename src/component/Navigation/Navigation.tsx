import React from "react";
import { NavLink } from "react-router-dom";

export const Navigation = (): JSX.Element => {
  return (
    <nav className={"navigation-bar"}>
      <div className={"btn home-btn"}>
        <NavLink exact to="/" activeClassName={"navigation-active"}>
          Home page
        </NavLink>
      </div>
      <div className={"btn list-btn"}>
        <NavLink to="/books" activeClassName={"navigation-active"}>
          List
        </NavLink>
      </div>
      <div className={"btn add-btn"}>
        <NavLink to="/addbook" activeClassName={"navigation-active"}>
          Add
        </NavLink>
      </div>
    </nav>
  );
};
