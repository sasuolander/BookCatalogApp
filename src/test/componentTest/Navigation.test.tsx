import React from 'react'
import { Navigation } from "./../../component/Navigation";
import { mount, shallow, render } from "enzyme";

it("render component Navigation",()=>{
    const component = shallow(<Navigation/>)
    expect(component).toMatchSnapshot();
})