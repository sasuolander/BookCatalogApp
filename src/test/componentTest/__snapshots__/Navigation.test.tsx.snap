// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component Navigation 1`] = `
<nav
  className="navigation-bar"
>
  <div
    className="btn home-btn"
  >
    <NavLink
      activeClassName="navigation-active"
      exact={true}
      to="/"
    >
      Home page
    </NavLink>
  </div>
  <div
    className="btn list-btn"
  >
    <NavLink
      activeClassName="navigation-active"
      to="/books"
    >
      List
    </NavLink>
  </div>
  <div
    className="btn add-btn"
  >
    <NavLink
      activeClassName="navigation-active"
      to="/addbook"
    >
      Add
    </NavLink>
  </div>
</nav>
`;
